src/modules/commander/mag_calibration.cpp

```diff
@@ -568,34 +599,46 @@ calibrate_return mag_calibrate_all(orb_advert_t *mavlink_log_pub)
 	// Calculate calibration values for each mag
 
 
-	float sphere_x[max_mags];
-	float sphere_y[max_mags];
-	float sphere_z[max_mags];
-	float sphere_radius[max_mags];
+	float sphere_x[max_mags] = {0.0f};
+	float sphere_y[max_mags] = {0.0f};
+	float sphere_z[max_mags] = {0.0f};
+	float sphere_radius[max_mags] = {0.2f};
```

Comment:
This only sets the first value, `sphere_radius = [0.2, 0.0, 0.0, ...]`
http://stackoverflow.com/questions/1065774/initialization-of-a-normal-array-with-one-default-value